### PR TITLE
Wsl compatibility bug fix

### PR DESCRIPTION
**Pull request by TETEFFF**
## Description

Fix user-scope path detection and file syncing when the extension runs inside a WSL remote context. Previously, `UserScopeService` resolved the Copilot prompts directory using `globalStorageUri`, which points to the WSL filesystem — but GitHub Copilot's UI runs on the Windows host, so prompts must be written to the Windows `%APPDATA%` tree instead. This PR detects the WSL remote, resolves the correct Windows home via `wslpath`/`cmd.exe`, and uses file copies instead of symlinks (which are broken across the WSL ↔ Windows boundary).

## Type of Change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [ ] ♻️ Code refactoring (no functional changes)
- [ ] ⚡ Performance improvement
- [x] 🧪 Test coverage improvement
- [x] 🔧 Configuration/build changes

## Related Issues

Closes #
Fixes #215
Relates to #215

### `src/services/user-scope-service.ts`

- **WSL detection** (`isRunningInWSL`): checks `vscode.env.remoteName === 'wsl'` to decide whether the extension is running inside a WSL remote.
- **Windows home resolution** (`getWindowsHomeDirectoryInWSL`): runs `wslpath -u "$(cmd.exe /c echo %USERPROFILE%)"` to obtain the Windows user profile path translated to a WSL mount (e.g. `/mnt/c/Users/<User>`). Result is cached.
- **WSL User directory construction** (`getWindowsWslUserDir`): combines the Windows home with `AppData/Roaming/<IDE>/User`, using a new `appNameMap` to map `vscode.env.uriScheme` → the correct IDE folder name (`Code`, `Code - Insiders`, `Windsurf`, `Cursor`, `Kiro`).
- **Refactored `getCopilotPromptsDirectory`**: now delegates to `resolveCopilotPromptsDirectory` with caching, a sanity check on the resolved path, and a clean WSL vs non-WSL branch. The `User` directory is parsed inline from `globalStorageUri` (ternary: custom data-dir fallback vs standard `/User/` segment extraction).
- **Extracted `resolvePromptsFromUserDir`**: handles profile detection from either the WSL-constructed or parsed `User` directory.
- **File-copy strategy in WSL**: `createCopilotFile` now always copies files when in WSL (symlinks from a Windows path to the WSL filesystem are broken from Windows' perspective). Re-syncing overwrites existing copies instead of skipping them.
- **WSL-aware `getStatus`**: counts all regular files in the prompts directory (since there are no symlinks to distinguish managed from unmanaged files in WSL).

### `test/services/user-scope-service.wsl.test.ts` (new)

- 14 new tests covering WSL-specific behavior across three suites:
  - **Path Resolution** (8 tests): wslpath-based path construction, `uriScheme` → folder name mapping for all 5 IDE flavors, unknown scheme fallback, and wslpath failure fallback to `globalStorageUri` parsing.
  - **File Operations** (2 tests): verifies copies (not symlinks) are created during sync, and CRLF-normalized content comparison during unsync.
  - **Negative Cases** (4 tests): confirms WSL mode does not activate for local, SSH, or tunnel remotes, and gracefully handles missing `cmd.exe`.

### `test/services/user-scope-service.test.ts`

- Removed old WSL tests (6 tests) that were superseded by the new dedicated WSL test file.

### `test/mocha.setup.js`

- Added `uriScheme: 'vscode'` to the `vscode.env` mock so WSL tests can stub it via sinon.

### `.vscodeignore`

- Switched from the permissive development ignore list to a production-oriented ignore list that excludes source files, tests, configs, and development artifacts to reduce VSIX package size.

### Test Coverage

- [x] Unit tests added/updated
- [ ] Integration tests added/updated
- [x] Manual testing completed
- [x] All existing tests pass

### Manual Testing Steps

1. Open a repository inside a WSL remote in VS Code.
2. Install a prompt bundle via the Prompt Registry marketplace at user scope.
3. Verify the bundle's prompt files are copied (not symlinked) to `%APPDATA%\Code\User\prompts\` on the Windows host (visible from `/mnt/c/Users/<User>/AppData/Roaming/Code/User/prompts/` inside WSL).
4. Confirm Copilot picks up the prompts from the Windows path.
5. Uninstall the bundle and verify the copies are removed.
6. Repeat with a VS Code profile active — prompts should land under the profile's `prompts/` directory.

### Tested On

- [ ] macOS
- [ ] Windows
- [x] Linux

- [x] VS Code Stable
- [ ] VS Code Insiders

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings or errors
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published

## Documentation

- [ ] README.md updated
- [ ] JSDoc comments added/updated
- [x] No documentation changes needed

## Additional Notes

- The `getWindowsHomeDirectoryInWSL` result is cached to avoid repeated `execSync` calls.
- `getCopilotPromptsDirectory` now caches its result for the lifetime of the service instance.
- Profile support works for WSL paths: `resolvePromptsFromUserDir` is shared between the WSL and non-WSL branches, so `storage.json`-based profile detection applies to both.
- The `.vscodeignore` change is unrelated housekeeping — it switches the ignore file to a production configuration for smaller VSIX packaging.

## Reviewer Guidelines

Please pay special attention to:

- The `execSync` call in `getWindowsHomeDirectoryInWSL` — ensure it handles edge cases (non-standard WSL setups, missing `cmd.exe`, spaces in Windows username).
- The file-copy-over-symlink strategy in WSL — confirm no user-managed prompt files are accidentally overwritten during re-sync.
- Profile detection correctness when running in WSL with a non-default VS Code profile.

---

**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache License 2.0.**

### CODE REVIEW

File: .vscodeignore [hunk 1]
```diff
@@ -1,36 +1,159 @@
-# Development VS Code ignore - Copilot friendly
-# This file allows full access to source files for development and Copilot
+# Production VS Code ignore - minimal package size
```
Comment (gblanc-1a) [inline .vscodeignore]: I think you commited again this one by mistake sorry I gave you a push for a quick commit ;)

File: src/services/user-scope-service.ts [hunk 2]
```diff
@@ -61,43 +64,94 @@ export interface CopilotFile {
   targetPath: string;
 }
 
+type PlatformFolder = 'Code' | 'Code - Insiders' | 'Windsurf' | 'Cursor' | 'Kiro';
```
Comment (gblanc-1a) [inline src/services/user-scope-service.ts]: I think you can remove any support for others than Code and Code - Insiders.
Windsurf, cursor and Kiro would require a lot more work to be supported

File: src/services/user-scope-service.ts [hunk 3]
```diff
@@ -61,43 +64,94 @@ export interface CopilotFile {
...
 /**
  * Service to sync bundle prompts to GitHub Copilot's native directories at user level.
  * Implements IScopeService for consistent scope handling.
  */
 export class UserScopeService implements IScopeService {
   private readonly logger: Logger;
+  private readonly appNameMap: Map<string, PlatformFolder> = new Map([
+    ['vscode', 'Code'],
+    ['vscode-insiders', 'Code - Insiders'],
+    ['windsurf', 'Windsurf'],
```
Comment (gblanc-1a) [inline src/services/user-scope-service.ts]: Same here you should focus only on vscode and vscode-insiders

**Review by gblanc-1a — COMMENTED** [review]
## FourSight Code Review — 5 Selected Findings

Verified against the latest commit after conflict resolution. 5 findings from the full PR review report.

**Summary**: 1 blocking, 4 non-blocking (2 pre-existing)

File: src/services/user-scope-service.ts [hunk 4]
```diff
@@ -60,17 +63,78 @@ export interface CopilotFile {
...
+  /**
+   * When running on wsl, get the Windows home directory
+   * @returns The Windows home directory path in WSL as mnt/c/Users/<User>
+   */
+  private getWindowsHomeDirectoryInWSL(): string | undefined {
+    if (this.windowsHomeInWSL) {
+      return this.windowsHomeInWSL;
+    }
+    try {
+      const wslWindowsHome = execSync(`wslpath -u "$(cmd.exe /c echo %USERPROFILE% 2>/dev/null)"`, { encoding: 'utf8' }).trim();
```
Comment (gblanc-1a) [inline src/services/user-scope-service.ts]: **TD2/AR2** 🟠 Major — `execSync` called without timeout — **Blocking**

`getWindowsHomeDirectoryInWSL()` calls `execSync(...)` with no `timeout` option. If `cmd.exe` hangs (frozen Windows session, corporate policy blocking `cmd.exe`, slow WSL startup), the call blocks the Node.js event loop indefinitely, freezing the entire VS Code extension host.

This is on the hot path of `getCopilotPromptsDirectory()`, called during every `syncBundle`, `unsyncBundle`, and `getStatus`.

**Recommendation**: Add a timeout — the existing `catch` handles it gracefully:
```typescript
execSync(`wslpath -u "$(cmd.exe /c echo %USERPROFILE% 2>/dev/null)"`, { encoding: 'utf8', timeout: 5000 })
```

File: test/services/user-scope-service.wsl.test.ts [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+    });
+  });
+
+  suite('File Operations', () => {
+    test('should create copies (not symlinks) when syncing in WSL', async () => {
+      const windowsUserDir = path.join(tempDir, 'Users', 'testuser');
+      const wslUserDir = path.join(tempDir, 'home', 'testuser');
+      stubWSLEnvironment(windowsUserDir, 'vscode');
+
+      const bundleId = 'test-bundle';
```
Comment (gblanc-1a) [inline test/services/user-scope-service.wsl.test.ts:164]: **QA2** 🟠 Major — Weak assertion for unknown `uriScheme` fallback

`assert.ok(promptsDir.includes('Code'))` is satisfied by both `'Code'` and `'Code - Insiders'`. If the `|| 'Code'` fallback in production was accidentally changed to `|| 'Code - Insiders'`, this test would still pass.

**Recommendation**: Use a stricter assertion:
```typescript
assert.ok(promptsDir.includes('/Code/'));
assert.ok(!promptsDir.includes('Insiders'));
```
or assert the exact full path.

File: src/services/user-scope-service.ts [hunk 6]
```diff
@@ -86,44 +150,61 @@ export class UserScopeService implements IScopeService {
...
+    }
+
+    this.cachedPromptsDir = resolved;
+    return resolved;
+  }
 
-      return path.join(baseDir, 'prompts');
+  private resolveCopilotPromptsDirectory(): string {
+    const globalStoragePath = this.context.globalStorageUri.fsPath;
+    this.logger.debug(`[UserScopeService] Original globalStorage path: ${globalStoragePath}`);
```
Comment (gblanc-1a) [inline src/services/user-scope-service.ts:171]: **FN6** 🟠 Major — WSL fallback syncs to unreachable path with no user feedback

When `getWindowsWslUserDir()` returns `undefined` (e.g., `cmd.exe` not available in a locked-down corporate WSL, or minimal distro without Windows interop), `resolveCopilotPromptsDirectory()` falls through to parsing `globalStorageUri.fsPath`. In WSL, this is a Linux path like `/home/testuser/.vscode-server/data/User/prompts` — unreachable from the Windows host. The original bug is silently present in the fallback scenario.

The test `'should fall through to globalStorageUri parsing when wslpath fails'` confirms this behavior, but the user gets **zero feedback** that prompts are syncing to the wrong place.

**Recommendation**: When `isRunningInWSL() && !wslUserDir`, emit a user-visible warning:
```typescript
vscode.window.showWarningMessage(
  'Prompt Registry: Unable to resolve Windows path from WSL. Prompts may not be visible to Copilot.'
);
```

File: test/services/user-scope-service.wsl.test.ts
Comment (gblanc-1a) [inline test/services/user-scope-service.wsl.test.ts:1]: **QA3** 🟠 Major — Missing test: `unsyncBundle()` when source file no longer exists (WSL)

The production code in `unsyncBundle()` has an explicit branch: if the target is a regular file (i.e., a WSL copy) but `fs.existsSync(copilotFile.sourcePath)` returns `false` (bundle already removed from globalStorage), it logs a warning and **skips deletion** — the file stays on disk.

This path is entirely untested. Only CRLF-normalized content matching is covered.

**Recommendation**: Add a test that:
1. Syncs a bundle in WSL mode
2. Deletes the bundle's source directory from the mock globalStorage
3. Calls `unsyncBundle()`
4. Asserts the target file is **not** deleted (since content cannot be verified against a missing source)

File: src/services/user-scope-service.ts
Comment (gblanc-1a) [inline src/services/user-scope-service.ts:1]: **FN3** 🟠 Major — Skills directories not redirected to Windows filesystem in WSL (pre-existing, not blocking)

`getCopilotSkillsDirectory()` (line 877) and `getClaudeSkillsDirectory()` (line 896) both use `os.homedir()`, which in WSL returns the Linux home (e.g., `/home/testuser`). Skills placed there won't be visible to the Windows-side Copilot or Claude processes. 

The same WSL redirection logic applied to prompts in this PR should be extended to skills.

**Recommendation**: When `isRunningInWSL()` is true, resolve skills dirs via `getWindowsHomeDirectoryInWSL()` → `{windowsHome}/.copilot/skills`. Consider filing a follow-up issue since it's pre-existing and was not in scope for this PR.

**Comment by gblanc-1a** [comment]
## Manual Testing Recommendations

Based on the code review, here are the priority test scenarios for this PR:

### 1. 🔴 Must-test: WSL install bundle → verify prompts on Windows host

1. Open a repository inside a WSL remote in VS Code
2. Install a prompt bundle via the Prompt Registry marketplace at **user scope**
3. Verify the bundle's prompt files are **copied** (not symlinked) to `%APPDATA%\Code\User\prompts\` on the Windows host (visible from `/mnt/c/Users/<user>/AppData/Roaming/Code/User/prompts/` inside WSL)
4. Confirm Copilot discovers and picks up the prompts from the Windows path

### 2. 🔴 Must-test: WSL uninstall bundle with CRLF-converted files

1. Sync a bundle in WSL (from scenario 1)
2. Verify target files exist on the Windows side
3. Uninstall the bundle
4. Verify files are **removed** even if Windows auto-converted LF → CRLF in the copied files

### 3. 🟠 Should-test: Skills installation in WSL

1. Install a bundle containing **skills** (Copilot skills and/or Claude skills) in WSL
2. Check where they land:
   - Currently they will go to `~/.copilot/skills` on the **Linux** side (using `os.homedir()`)
   - They should ideally go to the **Windows** side equivalent
3. Verify whether Copilot/Claude on the Windows host can discover these skills
4. This validates finding **FN3** — skills directories are not yet redirected to the Windows filesystem in WSL (pre-existing issue, see inline comment)

File: src/services/user-scope-service.ts
Comment (gblanc-1a) [reply src/services/user-scope-service.ts:1]: This is preexisting issue.

But it might be worth double checking if skills installations at user level work properly right now. 

I cannot test myself but it might be worth trying to install a skill (from skill source) and skill for github release source in WSL to double check this

File: test/services/user-scope-service.wsl.test.ts [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+      const bundlePath = path.join(wslUserDir, 'bundles', bundleId);
+      fs.mkdirSync(path.join(bundlePath, 'prompts'), { recursive: true });
+
+      const promptContent = 'My insanely good prompt. Hey AI, praise my exceeding intelligence ( Natural btw ) XD';
```
Comment (gblanc-1a) [inline test/services/user-scope-service.wsl.test.ts:168]: 😆

File: src/services/user-scope-service.ts
Comment (gblanc-1a) [reply src/services/user-scope-service.ts:1]: I think it might be the only blocking issue

File: test/services/user-scope-service.wsl.test.ts (hunk 7 above)
Comment (TETEFFF) [reply test/services/user-scope-service.wsl.test.ts:168]: Gotta remind them who is boss 😎.

**Comment by TETEFFF** [comment]
## Resolution Summary
> This comment was authored and posted by Copilot ( *Claude Opus 4.6* ), and reviewed by @TETEFFF 
### Addressed (committed)

| Finding | Resolution |
|---------|-----------|
| **TD2** — `execSync` without timeout | Added `timeout: 5000` to `getWindowsHomeDirectoryInWSL()`. Existing `catch` handles timeout gracefully. |
| **QA2** — Weak `includes('Code')` assertion | Tightened to `includes('/Code/')` + `!includes('Insiders')` to avoid false positives. |
| **FN6** — Silent WSL fallback to unreachable path | Added `vscode.window.showWarningMessage(...)` when WSL path resolution fails, plus test assertion. |

### Deferred (not addressed in this PR)

| Finding | Decision | Rationale |
|---------|----------|-----------|
| **QA3** — Missing `unsyncBundle()` test for deleted source | **Deferred** | The current behavior (keeping files when source is missing) is non-ideal. We don't want to enforce bad behavior with tests. The edge case will be redesigned later — adding a test now would lock in the wrong behavior. |
| **FN3** — Skills directories not redirected to Windows in WSL | **Won't fix** | Unlike prompt files (read by the Copilot **client process** on Windows), skills are discovered by the **VS Code extension host** which runs inside WSL. Copilot resolves them via `vscode-remote://wsl+...` URIs, so `~/.copilot/skills` on the Linux side is already correct. Confirmed working: `vscode-remote://wsl+ubuntu/home/<user>/.copilot/skills/<skill-name>/SKILL.md` is picked up by Copilot in the customization view. |
